File: backend_server/models/product.py
from sqlalchemy import Column, Integer, String, Float, Boolean
from sqlalchemy.orm import declarative_base

Base = declarative_base()

class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, unique=True, index=True, nullable=False)
    description = Column(String, default="")
    price = Column(Float, nullable=False)
    # For simplicity, we'll store an image URL. You could store base64 or a file path later.
    image_url = Column(String, default="")
    category = Column(String, default="Uncategorized")
    is_available = Column(Boolean, default=True)

File: mobile_app/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final String category;
  final bool isAvailable;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.category,
    required this.isAvailable,
  });

  // Convert JSON from API to Product object
  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'] ?? '',
      price: json['price'].toDouble(),
      imageUrl: json['image_url'] ?? '',
      category: json['category'] ?? 'Uncategorized',
      isAvailable: json['is_available'] ?? true,
    );
  }

  // Convert Product object to JSON for API calls
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'image_url': imageUrl,
      'category': category,
      'is_available': isAvailable,
    };
  }
}